polaris-core/src/main/java/org/apache/polaris/core/persistence/dao/entity/EntitiesResult.java

```diff
@@ -23,13 +23,21 @@
 import jakarta.annotation.Nonnull;
 import jakarta.annotation.Nullable;
 import java.util.List;
+import java.util.Optional;
 import org.apache.polaris.core.entity.PolarisBaseEntity;
+import org.apache.polaris.core.persistence.pagination.Page;
+import org.apache.polaris.core.persistence.pagination.PageToken;
 
 /** a set of returned entities result */
 public class EntitiesResult extends BaseResult {
 
   // null if not success. Else the list of entities being returned
   private final List<PolarisBaseEntity> entities;
+  private final Optional<PageToken> pageTokenOpt;
+
+  public static EntitiesResult fromPolarisPage(Page<PolarisBaseEntity> page) {
```

Comment:
Why "Polaris Page"? :)

Reply:
agh, good catch! I put too much faith in my IDE when refactoring `PolarisPage` -> `Page`.